### ISSUE TITLE
Broken on 32-bit platforms

### ISSUE THREAD

**Issue by ofek** [rustls/webpki#112]
Targets affected in our CI:

- `i686-unknown-linux-gnu`
- `i686-pc-windows-msvc`

```
   Compiling hmac v0.12.1
error[E0080]: evaluation of constant value failed
   --> /cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-webpki-0.101.0/src/der.rs:246:45
    |
246 | const LONG_FORM_LEN_FOUR_BYTES_MAX: usize = (1 << (8 * 4)) - 1;
    |                                             ^^^^^^^^^^^^^^ attempt to shift left by `32_i32`, which would overflow

note: erroneous constant used
   --> /cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-webpki-0.101.0/src/der.rs:211:40
    |
211 | pub(crate) const MAX_DER_SIZE: usize = LONG_FORM_LEN_FOUR_BYTES_MAX;
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0080`.
error: could not compile `rustls-webpki` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...

Compilation of failed (code 101)
```

**Comment by cpu** [comment]
Thanks for catching this. I'm raising the issue for discussion on what to do next but probably we will want to yank rustls-webpki 0.101.0 and rustls 0.21.3 until this is fixed and a new release cut.

**Comment by cpu** [comment]
We've published v0.101.1 to fix this.

**Comment by ofek** [comment]
Thank you very much!
